src/main/groovy/com/github/benmanes/gradle/versions/updates/Resolver.groovy

```diff
@@ -149,6 +159,19 @@ class Resolver {
     }
   }
 
+  /** Returns a variant of the provided dependency used for querying the latest version. */
+  @TypeChecked(SKIP)
+  private Dependency createQueryDependency(DependencyConstraint dependency, String revision) {
```

Comment:
Its probably not simpler to combine this with the other `createQueryDependency`, though the duplication could lead to mistakes of fixing one instead of both.